sqlglot/dialects/teradata.py

```diff
@@ -13,6 +13,30 @@
 from sqlglot.tokens import TokenType
 
 
+def _date_add_sql(
+    kind: t.Literal["+", "-"],
+) -> t.Callable[[Teradata.Generator, exp.DateAdd | exp.DateSub], str]:
+    def func(self: Teradata.Generator, expression: exp.DateAdd | exp.DateSub) -> str:
+        this = self.sql(expression, "this")
+        unit = expression.args.get("unit")
+
+        expression = self._simplify_unless_literal(expression.expression)
+        if not isinstance(expression, exp.Literal):
+            self.unsupported("Cannot add non literal")
+
+        this_sql = self.sql(expression, "this")
+        if this_sql.startswith("-"):
```

Comment:
this section doesn’t seems a bit suspect. you shouldn’t generate sql just to check if it’s a subtraction, just check the class of the ast node

Reply:
Agree it's not the nicest. What do you expect the class of the ast node to be? For a sql like `SELECT DATE_ADD('2023-01-01', -5, DAY)` on line 27 `expression` looks like `Literal(this=-5, is_string=False)`

Reply:
it shouldn't be that, it should be neg(literal)

```
python -m sqlglot "date_add('2020-01-01', -5)" --parse
DateAdd(
  this=Literal(this=2020-01-01, is_string=True),
  expression=Neg(
    this=Literal(this=5, is_string=False)))
```

Reply:
ah yup, the `_simplify_unless_literal` was folding the neg into the literal. I updated to check for the neg before that call